corehq/apps/custom_data_fields/static/custom_data_fields/js/custom_data_fields.js

```diff
@@ -1,163 +1,182 @@
-            'regex': regex,
-            'regex_msg': regex_msg,
-            'is_multiple_choice': is_multiple_choice,
-            'index_in_fixture': self.index_in_fixture(),
+        self.submitFields = function (fieldsForm) {
+            var customDataFieldsForm = $("<form>")
+                .attr("method", "post")
+                .attr("action", fieldsForm.action);
+
+
+            $("<input type='hidden'>")
+                .attr("name", 'csrfmiddlewaretoken')
+                .attr("value", $("#csrfTokenContainer").val())
+                .appendTo(customDataFieldsForm);
+
+            $('<input type="hidden">')
+                .attr('name', 'data_fields')
+                .attr('value', JSON.stringify(self.serialize()))
+                .appendTo(customDataFieldsForm);
+
+            $('<input type="hidden">')
+                .attr('name', 'purge_existing')
+                .attr('value', self.purge_existing())
+                .appendTo(customDataFieldsForm);
+            customDataFieldsForm.appendTo("body");
+            customDataFieldsForm.submit();
         };
-    };
-}
-
-
-function CustomDataFieldsModel () {
-    var self = this;
-    self.data_fields = ko.observableArray();
-    self.purge_existing = ko.observable(false);
-    // The data field that the "remove field modal" currently refers to.
-    self.modalField = ko.observable();
-
-    self.addField = function () {
-        self.data_fields.push(new CustomDataField());
-    };
-
-    self.removeField = function (field) {
-        self.data_fields.remove(field);
-    };
-
-    self.setModalField = function (field) {
-        self.modalField(field);
-    };
-
-    self.confirmRemoveField = function () {
-        // Remove the field that the "remove field modal" currently refers to.
-        self.removeField(self.modalField());
-    };
-
-    self.init = function (initialFields) {
-        _.each(initialFields, function (field) {
-            custom_field = new CustomDataField();
-            custom_field.init(field);
-            self.data_fields.push(custom_field);
-            custom_field.choices.subscribe(function() {
-                $("#save-custom-fields").prop("disabled", false);
-            });
+    }
+
+    $(function () {
+        var customDataFieldsModel = new CustomDataFieldsModel();
+        customDataFieldsModel.init(hqImport('hqwebapp/js/initial_page_data').get('custom_fields'));
+        customDataFieldsModel.data_fields.subscribe(function() {
+            $("#save-custom-fields").prop("disabled", false);
         });
-    };
-
-    self.serialize = function () {
-        var fields = [];
-        var fieldsToRemove = [];
-        _.each(self.data_fields(), function (field) {
-            if(field.slug() || field.label()) {
-                fields.push(field.serialize());
-            } else {
-                fieldsToRemove.push(field);
-            }
+
+        ko.applyBindings(customDataFieldsModel, $('#custom-fields-form').get(0));
```

Comment:
fyi, you can change this to `$('#custom-fields-form').koApplyBindings(customDatFieldsModel)` to get a check that the element being passed exists (if `ko.applyBindings` gets passed something falsy, it applies the bindings to the entire page, which is an occsional source of bugs).